website/project/model.py

```diff
@@ -149,6 +149,31 @@ class MetaData(GuidStoredObject):
     date_modified = fields.DateTimeField(auto_now=datetime.datetime.utcnow)
 
 
+def validate_contributor(guid, contributors):
+    if guid != '':
+        user = User.find(
+            Q('_id', 'eq', guid) &
+            Q('is_claimed', 'eq', True)
+        )
+        if user.count() != 1:
+            raise ValidationValueError('User does not exist or is not active.')
+        elif guid not in contributors:
+            raise ValidationValueError('Mentioned user is not a contributor.')
+    return True
+
+def add_new_mentions(comment, contributors):
+        """ Check if there are any new_mentions that are not in old_mentions and are a contributor
+
+        :param Node comment: Node that has new_mentions and old_mentions
+        :param list contributors: List of contributors on the node
+        :return bool new_mentions_added: Whether there are valid new_mentions
+        """
+        validate = lambda m: m not in comment.old_mentions and validate_contributor(m, contributors)
+        comment.new_mentions = filter(validate, comment.new_mentions)
+        if len(comment.new_mentions) > 0:
+            return True
```

Comment:
Empty lists are false-y in Python, so we can simplify this down to:

```
return bool(comment.new_mentions)
```

However, that'll introduce a subtle portability bug, since `filter` doesn't return a list in Python 3. I suppose that's a reason to go with a list comprehension instead of `filter` after all (cf. https://github.com/CenterForOpenScience/osf.io/pull/5376#discussion_r59867266).

```
$ python2
Python 2.7.11 (default, Mar 18 2016, 15:44:22) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> bool(filter(lambda a: a, []))
False
>>> bool([a for a in []])
False
>>>
```

```
>>>$ python3
Python 3.5.0 (default, Mar 18 2016, 15:49:36) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.72)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> bool(filter(lambda a: a, []))
True
>>> bool([a for a in []])
False
>>>
```